### ISSUE TITLE
MPD protocol: "noidle" command returns "OK" even when not in idle

### ISSUE THREAD

**Issue by geneticdrift** [owntone/owntone-server#1611]
owntone 28.4
Debian 4.19.282-1

MPD ignores "noidle" commands unless the client connection is already in "idle" state.
Owntone responds with "OK" to "noidle" even when the connection is not in "idle" state.

To reproduce:
```
# this is owntone response to noidle
echo "noidle" | nc -N 192.168.122.149 6603
OK MPD 0.20.0
OK
```
```
# this is MPD response to noidle
echo "noidle" | nc -N localhost 6601
OK MPD 0.24.0
```

This may cause MPD clients to not be able to keep track of the response to request sequence.
A client needs to send "noidle" when the last command it sent was "idle" to interrupt it and send another command.
But it is possible that the client connection in MPD has already sent a response to the last "idle" that the client has not yet received. So MPD must ignore the noidle command unless the client connection is still in idle state.

**Comment by ejurgensen** [comment]
The maintainer of the mpd part of owntone isn't active any more. I can make a fix if it is a simple one, but I'm not myself well acquiented with the protocol, and I didn't completely understand your description. Could you spell out for me what it is owntone should or shouldn't do?

**Comment by ejurgensen** [comment]
That would be awesome, sounds like you know this stuff

**Comment by grobian** [comment]
isn't this just an `else if (ctx->is_idle)` instead of `else` in mpd_command_noidle to fix this?

**Comment by ejurgensen** [comment]
Fixed in PR #1794
